engines/suppliers/app/controllers/suppliers/api/v1/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Suppliers
+  module Api
+    module V1
+      class SuppliersController < ApplicationController
+        before_action :load_supplier, only: [:show, :update, :destroy]
+
+        # GET /api/v1/suppliers
+        #
+        def index
+          suppliers = SuppliersFinder.new(supplier_params[:group_id]).find
```

Comment:
Absolutely, this is what I actually have in mind, but I think that for now we can postpone this step to see how far we get. We'll see how it fits after we add all the CRUDs. Anyway, the mantra is to keep responsibilities away of the model.